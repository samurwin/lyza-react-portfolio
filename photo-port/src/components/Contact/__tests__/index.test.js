import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContactForm from '..';

afterEach(cleanup)

describe('Contact Form', () => {
    it('renders', () => {
        render(<ContactForm></ContactForm>);
    })

    it('matches snapshot', () => {
        const { asFragment } = render(<ContactForm></ContactForm>);

        expect(asFragment()).toMatchSnapshot();
    })

    it('h1 tag renders', () => {
        const { getByTestId } = render(<ContactForm></ContactForm>);

        expect(getByTestId('contact')).toHaveTextContent('Contact Me');
    })

    it('button renders', () => {
        const { getByTestId } = render(<ContactForm></ContactForm>);

        expect(getByTestId('btn')).toHaveTextContent('Submit');
    })
})