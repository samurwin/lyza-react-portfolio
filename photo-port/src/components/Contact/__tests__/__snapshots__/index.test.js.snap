// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Form matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="contact"
    >
      Contact Me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        data-testid="btn"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
